### PR TITLE
feat: Update customer related typings and sdk methods

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

File: packages/core/types/src/http/customer/common.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  address_1: string | null
+  address_2: string | null
+  city: string | null
+  country_code: string | null
+  province: string | null
+  postal_code: string | null
+  phone: string | null
+  metadata: Record<string, unknown> | null
+  created_at: string
+  updated_at: string
```
Comment (adrien2p) [inline packages/core/types/src/http/customer/common.ts:30]: Q: isn't it always Date | string?

File: packages/medusa/src/api/store/orders/query-config.ts [hunk 2]
```diff
@@ -59,5 +59,5 @@ export const retrieveTransformQueryConfig = {
 
 export const listTransformQueryConfig = {
   defaults: defaultStoreOrderFields,
-  isList: false,
+  isList: true,
```
Comment (adrien2p) [inline packages/medusa/src/api/store/orders/query-config.ts:62]: nice catch 🙏

**Review by adrien2p — APPROVED** [review]
We are getting there 💪

File: packages/core/types/src/http/customer/common.ts (hunk 1 above)
Comment (sradevski) [reply packages/core/types/src/http/customer/common.ts:30]: @adrien2p I think when sending to the BE it first needs to be serialized to string, and the response will also need to be parsed from string to Date, that doesn't happen automatically currently.

**Pull request merged**